janitor/functions.py

```diff
@@ -2289,6 +2293,7 @@ def row_to_names(
     row_number: int = None,
     remove_row: bool = False,
     remove_rows_above: bool = False,
+    reset_index: bool = True,
```

Comment:
@fireddd I think @jk3587 brings up a good point and that we should set the default value of `reset_index` to `False`.

This isn't because I disagree with the reasons that you stated. It's just that I think we should avoid making breaking changes without warning users.

The best course of action here, then, is to set the value to `False` and raise a warning for users that says something like:

```python
warnings.warn(
    "The function row_to_names will, in the official 1.0 release, "
    "change its behaviour to reset the dataframe's index by default. "
    "You can prepare for this change right now by explicitly setting "
    "`reset_index=True` when calling on `row_to_names`."
)
```

(If you copy/paste the message verbatim, please preserve the spacing just before the quotes!)